strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProvider.java

```diff
@@ -67,74 +68,38 @@ public String getAlias()
+                                   ArtifactTransportException,
+                                   ProviderImplementationException
+    {   
+        RepositoryPath artifactPath = getPath(storageId, repositoryId, path);
+        
+        if(artifactPath == null)
         {
-            return is;
-        }
-
-        for (String storageAndRepositoryId : groupRepository.getGroupRepositories())
-        {
-            String sId = getConfigurationManager().getStorageId(storage, storageAndRepositoryId);
-            String rId = getConfigurationManager().getRepositoryId(storageAndRepositoryId);
-
-            Repository r = getConfiguration().getStorage(sId).getRepository(rId);
-
-            if (!r.isInService())
-            {
-                continue;
-            }
-            if (artifactRoutingRulesChecker.isDenied(repositoryId, rId, artifactPath))
-            {
-                continue;
-            }
-            try
-            {
-                is = resolveArtifact(sId, r.getId(), artifactPath);
-            }
-            catch (IOException e)
-            {
-                continue;
-            }
-            if (is != null)
-            {
-                return is;
-            }
-        }
-
-        return null;
-    }
-
-    private ArtifactInputStream resolveDirectlyFromGroupPathIfPossible(final String storageId,
-                                                                       final String repositoryId,
-                                                                       final String path)
-            throws IOException
-    {
-
-        final Storage storage = getConfiguration().getStorage(storageId);
-        final Repository repository = storage.getRepository(repositoryId);
-        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
-        final RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
-        final RepositoryFileAttributes artifactFileAttributes;
-        try
-        {
-            artifactFileAttributes = Files.readAttributes(artifactPath, RepositoryFileAttributes.class);
+            logger.debug("Could not resolve path to artifact in Group Repository");
+            return null;
         }
-        catch (NoSuchFileException ex)
+                        
+        logger.debug("Resolved path for Group Repository = " + artifactPath);
+        
+        Repository repository = artifactPath.getFileSystem().getRepository();
+        String resolvedStorageId = repository.getStorage().getId();
+        String resolvedRepositoryId = repository.getId();
+        String resolvedPath = artifactPath.relativize().toString();
+        RepositoryProvider provider = getRepositoryProviderRegistry().getProvider(repository.getType());
+        
+        //If current repository is a group repository present in local cache
+        if(getAlias().equals(provider.getAlias()) && Files.exists(artifactPath))
```

Comment:
we can't have artifact files physically stored in group repository folder, so this condition will never be true